src/zope/i18n/compile.py

```diff
@@ -1,54 +1,61 @@
-from contextlib import closing
+
 import logging
-import os
-from os.path import join
-from stat import ST_MTIME
+logger = logging.getLogger('zope.i18n')
+
+def _cannot_compile_mo_file(domain, lc_messages_path):
+    logger.critical("Unable to compile messages: Python `gettext` library missing.")
+    return
 
-HAS_PYTHON_GETTEXT = True
 try:
     from pythongettext.msgfmt import Msgfmt
     from pythongettext.msgfmt import PoSyntaxError
-except ImportError:
+except ImportError: # pragma: no cover
     HAS_PYTHON_GETTEXT = False
+    compile_mo_file = _cannot_compile_mo_file
+else:
+    from contextlib import closing
+    import os.path
+    from os.path import join
 
-logger = logging.getLogger('zope.i18n')
+    HAS_PYTHON_GETTEXT = True
 
+    def _safe_mtime(path):
+        try:
+            return os.path.getmtime(path)
+        except (IOError, OSError):
+            return None
 
-def compile_mo_file(domain, lc_messages_path):
-    """Creates or updates a mo file in the locales folder."""
-    if not HAS_PYTHON_GETTEXT:
-        logger.critical("Unable to compile messages: Python `gettext` library missing.")
-        return
+    def compile_mo_file(domain, lc_messages_path):
+        """Creates or updates a mo file in the locales folder."""
 
-    base = join(lc_messages_path, domain)
-    pofile = str(base + '.po')
-    mofile = str(base + '.mo')
+        base = join(lc_messages_path, domain)
+        pofile = str(base + '.po')
+        mofile = str(base + '.mo')
 
-    po_mtime = 0
-    try:
-        po_mtime = os.stat(pofile)[ST_MTIME]
-    except (IOError, OSError):
-        return
+        po_mtime = _safe_mtime(pofile)
+        mo_mtime = _safe_mtime(mofile) if os.path.exists(mofile) else 0
```

Comment:
It seems strange to me to distinguish the cases where the file doesn't exist and where it existed briefly but was gone in the next nanosecond.

Reply:
Agreed. That's just what it was doing before.